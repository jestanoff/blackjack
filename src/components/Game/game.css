.board--controls {
  margin: 20px 0;
  padding: 10px;
  background: rgba(0, 0, 0, 0.25);
  border-radius: 4px;
}

.board--controls-hr {
  border: none;
  border-top: 1px solid rgba(0, 0, 0, 0.25);
  border-bottom: 1px solid rgba(255, 255, 255, 0.25);
  margin: 10px -10px;
}

.board--ai-cards,
.board--player-cards {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: auto;
  margin-bottom: 20px;
}
