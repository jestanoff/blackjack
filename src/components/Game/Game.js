import React, { Component } from 'react'
import { Button, Card, Status, Text } from '../'
import {
  getFullDeck,
  getRandomCard,
  getUniqueCard,
  removeCardFromDeck,
  sumCards,
} from '../../utils'
import './game.css'

class Game extends Component {
  constructor(props) {
    super(props)

    this.state = this.getInitialState()
  }

  getInitialState = () => {
    let deckStatus = getFullDeck()
    const firstPlayerCard = getRandomCard()
    deckStatus = removeCardFromDeck(firstPlayerCard, deckStatus)
    const secondPlayerCard = getUniqueCard(deckStatus)
    deckStatus = removeCardFromDeck(secondPlayerCard, deckStatus)
    const firstAiCard = getUniqueCard(deckStatus)

    return {
      deckStatus,
      playerCards: [firstPlayerCard, secondPlayerCard],
      aiCards: [firstAiCard],
    }
  }

  dealCard = event => {
    event.preventDefault()
  }

  handleResetGame = event => {
    event.preventDefault()
    const state = this.getInitialState()
    this.setState(state)
  }

  render() {
    const { aiCards, playerCards } = this.state
    const sumPlayer = sumCards(playerCards)
    const sumAi = sumCards(aiCards)

    return (
      <section className="board">
        <section className="board--cards">
          <div className="board--ai-cards">
            <div>
              <h2>Dealer cards</h2>
              <p>sum {sumAi}</p>
            </div>
            {aiCards.map(({ suit, value }) => (
              <Card key={`${value}-${suit}`} suit={suit} value={value} />
            ))}
          </div>
          <div className="board--player-cards">
            <div>
              <h2>Player cards</h2>
              <p>sum {sumPlayer}</p>
            </div>
            {playerCards.map(({ suit, value }) => (
              <Card key={`${value}-${suit}`} suit={suit} value={value} />
            ))}
          </div>
        </section>
        <section className="board--controls">
          <Status sumPlayer={sumPlayer} sumAi={sumAi} />
          <hr className="board--controls-hr" />
          <Button handleClick={this.dealCard} color="tertiary">
            Hit
          </Button>
          <Button color="primary">Stick</Button>
          <hr className="board--controls-hr" />
          <Button color="warning" handleClick={this.handleResetGame}>
            Reset
          </Button>
        </section>
      </section>
    )
  }
}

export default Game
