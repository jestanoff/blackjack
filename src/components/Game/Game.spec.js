import React from 'react'
import Game from './Game'
import { shallow } from 'enzyme'

describe.skip('<Game />', () => {
  it('should render proper markup', () => {
    const wrapper = shallow(<Game />)

    expect(wrapper.find('board')).toBeDefined()
  })
})
