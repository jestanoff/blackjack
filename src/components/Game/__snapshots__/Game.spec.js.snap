// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> should render proper markup 1`] = `
<section
  className="board"
>
  <section
    className="board--cards"
  >
    <div
      className="board--ai-cards"
    >
      <div>
        <h2>
          AI cards:
        </h2>
        <p>
          sum 
          10
        </p>
      </div>
      <div
        className="card card-suit-heart card-value-Q"
      />
    </div>
    <div
      className="board--player-cards"
    >
      <div>
        <h2>
          Player cards:
        </h2>
        <p>
          sum 
          17
        </p>
      </div>
      <div
        className="card card-suit-diamond card-value-7"
      />
      <div
        className="card card-suit-club card-value-K"
      />
    </div>
  </section>
  <section
    className="board--controls"
  >
    <h1
      className="text text-color-dark-grey text-size-standard"
    >
      Players turn
    </h1>
    <hr
      className="board--controls-hr"
    />
    <button
      className="button button-color-tertiary"
      onClick={[Function]}
    >
      Hit
    </button>
    <button
      className="button button-color-primary"
      onClick={undefined}
    >
      Stick
    </button>
    <hr
      className="board--controls-hr"
    />
    <button
      className="button button-color-warning"
      onClick={[Function]}
    >
      Reset
    </button>
  </section>
</section>
`;
